packages/compartment-mapper/src/import-hook.js

```diff
@@ -196,10 +196,16 @@ export const makeImportHookMaker = (
             record = {
               record: concreteRecord,
               specifier: candidateSpecifier,
+              meta: { url: moduleLocation },
             };
           } else {
-            record = concreteRecord;
+            record = {
+              record: concreteRecord,
+              specifier: moduleSpecifier,
+              meta: { url: moduleLocation },
+            };
           }
+          // ^ to be simplified, left unchanged for a more comprehensive diff
```

Comment:
This and the “draft” status suggests to me you’re not looking for a stamp yet. I’ll leave comments for now.